kernel/src/syscall/fcntl.rs

```diff
@@ -154,6 +157,39 @@ fn handle_setown(fd: FileDesc, arg: u64, ctx: &Context) -> Result<SyscallReturn>
     Ok(SyscallReturn::Return(0))
 }
 
+fn handle_addseal(fd: FileDesc, arg: u64, ctx: &Context) -> Result<SyscallReturn> {
+    let new_seals = FileSeals::from_bits(arg as u32)
+        .ok_or_else(|| Error::with_message(Errno::EINVAL, "invalid seals"))?;
+
+    let mut file_table = ctx.thread_local.borrow_file_table_mut();
+    let file = get_file_fast!(&mut file_table, fd);
+    let memfd_file = file.downcast_ref::<MemfdFile>().ok_or_else(|| {
+        Error::with_message(
+            Errno::EINVAL,
+            "file seals can only be applied to memfd files",
+        )
+    })?;
+
+    memfd_file.memfd_inode().add_seals(new_seals)?;
```

Comment:
https://github.com/google/gvisor/blob/feddee9574d7d57fffb79d5b3a31ac85927ba72e/test/syscalls/linux/memfd.cc#L504-L510

So I think we should still have `memfd_file.add_seals()`/`memfd.get_seals()` instead of exposing `memfd_inode` directly?